```
======================================================================
FAIL: test_context_manager_restores_backup (__main__.TestRestorableFileWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/geopm/geopm/service/geopmdpy_test/TestRestorableFileWriter.py", line 134, in test_context_manager_restores_backup
    self.assertCountEqual([], warnings)
AssertionError: Element counts were not equal:
First has 0, Second has 1:  "Encountered exception in secure read of /tmp/tmpqk4_98g_TestRestorableFileWriter/__main__.TestRestorableFileWriter.test_context_manager_restores_backup.backup: [Errno 13] Permission denied: '/tmp/tmpqk4_98g_TestRestorableFileWriter/__main__.TestRestorableFileWriter.test_context_manager_restores_backup.backup'"
First has 0, Second has 1:  'Unable to securely read /tmp/tmpqk4_98g_TestRestorableFileWriter/__main__.TestRestorableFileWriter.test_context_manager_restores_backup.backup'

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
test_context_manager_warns_on_cleanup_failure (__main__.TestRestorableFileWriter) ... FAIL

======================================================================
FAIL: test_context_manager_warns_on_cleanup_failure (__main__.TestRestorableFileWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/geopm/geopm/service/geopmdpy_test/TestRestorableFileWriter.py", line 156, in test_context_manager_warns_on_cleanup_failure
    self.assertEqual(1, len(warnings))
AssertionError: 1 != 2

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
test_modify_does_not_overwrite_existing_backup (__main__.TestRestorableFileWriter) ... ERROR

======================================================================
ERROR: test_modify_does_not_overwrite_existing_backup (__main__.TestRestorableFileWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/geopm/geopm/service/geopmdpy_test/TestRestorableFileWriter.py", line 74, in test_modify_does_not_overwrite_existing_backup
    with open(self._backup_path) as f:
PermissionError: [Errno 13] Permission denied: '/tmp/tmpz74w5h2mTestRestorableFileWriter/__main__.TestRestorableFileWriter.test_modify_does_not_overwrite_existing_backup.backup'

----------------------------------------------------------------------


```
Seems to be trying to create a file in /tmp.  We would like these temporary files created in /run/geopm/ so that they cannot be listed.

--- a/service/geopmdpy/system_files.py
+++ b/service/geopmdpy/system_files.py
@@ -30,6 +30,7 @@ import grp
 import pwd
 import fcntl
 import subprocess # nosec
+import tempfile
 from functools import lru_cache
 
 from . import pio
@@ -161,12 +162,14 @@ def secure_make_file(path, contents):
 
     The file is created by first writing the output to a temporary
     file with restricted permissions (mode 0o600) within the same
-    directory, and then renaming the file after it is closed.  This
-    temporary file with have the path `<path>-<uuid>-tmp`.
+    directory, and then renaming the file after it is closed.
 
     The rename operation is atomic, and will overwrite any existing
     file located in the specified path.  The permissions on the output
     file are mode 0o600 with uid and gid matching the process values.
+
+    The temporary file has the prefix `tmp.`. This file may persist if
+    the program exits abruptly.
 
     Args:
         path (str): The path where the file is created
@@ -174,12 +177,13 @@ def secure_make_file(path, contents):
         contents (str): The contents of the created file
 
     """
-    temp_path = f'{path}-{uuid.uuid4()}-tmp'
+    directory = os.path.dirname(os.path.abspath(path))
+    prefix = f'{os.path.basename(path)}-tmp-'
     old_mask = os.umask(0o077)
     try:
-        with open(os.open(temp_path, os.O_CREAT | os.O_WRONLY, 0o600), 'w') as file:
+        with tempfile.NamedTemporaryFile(mode="w+", prefix=prefix, dir=directory, delete=False) as file:
             file.write(contents)
-        os.rename(temp_path, path)
+        os.rename(file.name, path)
     finally:
         os.umask(old_mask)
 

